### ISSUE TITLE
FINNGEN_R5_I9_HEARTFAIL_AND_CHD causes internal server error in coloc query

### ISSUE THREAD

**Issue by ktsirigos** [opentargets/issues#2585]
User reported an error in one of our entries in the Genetics Portal - shows 'Internal server error':

https://genetics.opentargets.org/gene/ENSG00000112137

**Comment by d0choa** [comment]
At least one study in the coloc results is breaking the API query 

Query:

```
query{
  colocalisationsForGene(geneId: "ENSG00000112137") {
    study {
      studyId
    }
  }
}
```

Response:
```
{
  "data": null,
  "errors": [
    {
      "message": "Internal server error",
      "path": [
        "colocalisationsForGene",
        41,
        "study"
      ],
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ]
    }
  ]
}
```

**Comment by JarrodBaker** [comment]
Thanks @d0choa! The API spec tells me that every  `colocalisationsForGene` entry is associated with one study: is this a correct description of the relationship?

**Comment by d0choa** [comment]
yes, every coloc is a comparison variantX, studyA, variantY, studyB. The `studyId` should always be there, but I wouldn't be surprised if the data has unexpected values.

**Comment by JarrodBaker** [comment]
The immediate cause of this error is that there is a study in the `v2d_coloc` table which is not present in the `studies` table: name, "FINNGEN_R5_I9_HEARTFAIL_AND_CHD".

Clickhouse query:
```
select distinct study_id from (
select left_study
from ot.v2d_coloc
where right_gene_id = 'ENSG00000112137' and right_type != 'gwas' and left_type = 'gwas') as s
left anti join studies on s.left_study = study_id;
```
The API crashes because it assumes that each study returned from the coloc table can be resolved to a study in the studies table.

**Comment by d0choa** [comment]
The lack of this study (`FINNGEN_R5_I9_HEARTFAIL_AND_CHD`) was also reported here:
https://github.com/opentargets/issues/issues/2073

IMO this is a data issue related to the v2d pipeline. Reassigning...

**Comment by DSuveges** [comment]
Luckily this is the only study that is missing from the final study index:
`In [15]: study_pre_df.join(study_df, on='study_id', how='left_anti').show(truncate=False, vertical=True)`

```
-RECORD 0-----------------------------------------------
 study_id             | FINNGEN_R5_I9_HEARTFAIL_AND_CHD 
 pmid                 |                                 
 pub_date             | 2021-5-11                       
 pub_journal          |                                 
 pub_title            |                                 
 pub_author           | FINNGEN_R5                      
 trait_reported       |                                 
 trait_efos           | [EFO_0000319]                   
 ancestry_initial     | [European=206656]               
 ancestry_replication | []                              
 n_initial            | 206656                          
 n_replication        | 0                               
 n_cases              | 8876                            
 trait_category       | Cardiovascular                  
 num_assoc_loci       | 3                               
 has_sumstats         | true       
```

When looking at the ETL code, most likely [this is](https://github.com/opentargets/genetics-pipe/blob/3eed6d1cf084192e6b6f34391f01bdb90464c961/src/main/scala/ot/geckopipe/index/V2DIndex.scala#L145) the condition that drops out this study:

```
.filter($"trait_reported".isNotNull)
```

For some reason, this study doesn't have reported trait, which doesn't make much sense. Interestingly, when looking at the pre-ELT study table, the reported trait is not null, it seems to be an empty string:

```
In [19]: study_pre_df.filter(f.col('trait_reported') == '').show(vertical=True)
-RECORD 0------------------------------------
 study_id             | FINNGEN_R5_I9_HEA... 
 pmid                 |                      
 pub_date             | 2021-5-11            
 pub_journal          |                      
 pub_title            |                      
 pub_author           | FINNGEN_R5           
 trait_reported       |                      
 trait_efos           | [EFO_0000319]        
 ancestry_initial     | [European=206656]    
 ancestry_replication | []                   
 n_initial            | 206656               
 n_replication        | 0                    
 n_cases              | 8876                 
 trait_category       | Cardiovascular       
 num_assoc_loci       | 3                    
 has_sumstats         | true  
```
vs
```
In [20]: sudy_pre_df.filter(f.col('trait_reported').isNull()).show(vertical=True)
(0 rows)
```

In the [buildStudiesIndex](https://github.com/opentargets/genetics-pipe/blob/3eed6d1cf084192e6b6f34391f01bdb90464c961/src/main/scala/ot/geckopipe/index/V2DIndex.scala#L109) function there is no other condition, restricted join that would explain the drop.

**Comment by DSuveges** [comment]
This issue is coming from the [FINNGEN manifest](https://r6.finngen.fi/api/phenos):

```json
{
  "assoc_files": [
    "/mnt/nfs/pheweb/r6/generated-by-pheweb/pheno_gz/I9_HEARTFAIL_AND_CHD.gz"
  ],
  "category": "Cardiometabolic endpoints",
  "category_index": 10,
  "gc_lambda": {
    "0.001": 1.1242,
    "0.01": 1.0844,
    "0.1": 1.0751,
    "0.5": 1.0718
  },
  "num_cases": 11034,
  "num_cases_prev": 8876,
  "num_controls": 234698,
  "num_gw_significant": 9,
  "num_gw_significant_prev": 3,
  "phenocode": "I9_HEARTFAIL_AND_CHD",
  "phenostring": ""
}
```

**Comment by d0choa** [comment]
Coalesce "phenostring" and "phenocode" to build the reported trait?

**Comment by ireneisdoomed** [comment]
Thank you everyone for the investigation and @DSuveges for the fix!

This is now fixed on [#19](https://github.com/opentargets/genetics-v2d-data/pull/19). 
As proposed by @d0choa, the reported trait has been successfully coalesced with the phenotype code so that the ETL will not drop this record when filtering out those where trait_reported == null

```python3
>>> studies[studies['study_id'] == 'FINNGEN_R6_I9_HEARTFAIL_AND_CHD'].iloc[0]
study_id                FINNGEN_R6_I9_HEARTFAIL_AND_CHD
pmid
pub_date                                     2022-01-24
pub_journal
pub_title
pub_author                                   FINNGEN_R6
trait_reported                     I9_HEARTFAIL_AND_CHD
trait_efos                                         None
ancestry_initial                      [European=245732]
ancestry_replication                                 []
n_initial                                        245732
n_replication                                       0.0
n_cases                                         11034.0
num_assoc_loci                                        0
has_sumstats                                       True
```
